internal/controller/pipelinerollout_controller.go

```diff
@@ -338,15 +338,25 @@ func applyPipelineSpec(
 	restConfig *rest.Config,
 	obj *kubernetes.GenericObject,
 	pipelineRollout *apiv1.PipelineRollout,
+	rolloutChildOp RolloutChildOperation,
+	childResouceSpecHash string,
 ) error {
 	numaLogger := logger.FromContext(ctx)
-	// TODO: use UpdateSpec instead
-	err := kubernetes.ApplyCRSpec(ctx, restConfig, obj, "pipelines")
-	if err != nil {
-		numaLogger.Errorf(err, "failed to apply Pipeline: %v", err)
-		pipelineRollout.Status.MarkFailed("ApplyPipelineFailure", err.Error())
-		return err
+
+	if rolloutChildOp != RolloutChildNoop {
```

Comment:
We can avoid pass in `RolloutChildOperation` here by checking `RolloutChildNoop` at L171, right?